chia/cmds/param_types.py

```diff
@@ -14,6 +14,48 @@
 from chia.wallet.util.address_type import AddressType
 
 
+def validate_uint64(
+    value: str,
+    fail_func: Callable[[str, Optional[click.Parameter], Optional[click.Context]], None],
+    param: Optional[click.Parameter],
+    ctx: Optional[click.Context],
+) -> uint64:
+    try:
+        d_value = Decimal(value)
+    except InvalidOperation:
+        fail_func("Value must be a valid uint64 number", param, ctx)
+    if d_value.is_signed():
+        fail_func("Value must be a positive integer", param, ctx)
+    if d_value != d_value.to_integral():
+        fail_func("Value must be an integer", param, ctx)
```

Comment:
why do we care about `d_value` here? afaict, we only return `u_value`

Reply:
for clearer error codes